src/core/sys/posix/aio.d

```diff
@@ -0,0 +1,66 @@
+/**
+ * D header file to interface with the
+ * $(HTTP pubs.opengroup.org/onlinepubs/9699919799/basedefs/aio.h.html, Linux aio API).
+ * Available since Linux 2.6
+ *
+ * Copyright: Copyright D Language Foundation 2018.
+ * License:   $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Authors:   $(HTTPS github.com/darredevil, Alexandru Razvan Caciulescu)
+ */
+module core.sys.posix.aio;
+
+private import core.sys.posix.signal;
+
+version (CRuntime_Glibc):
+version (X86_64):
+
+extern (C):
+@system:
+@nogc:
+nothrow:
+
+struct aiocb
+{
+    int aio_fildes;
+    int aio_lio_opcode;
+    int aio_reqprio;
+    void *aio_buf;   //volatile
+    size_t aio_nbytes;
+    sigevent aio_sigevent;
+
+    ubyte[24] internal_members_padding;
+    off_t aio_offset;
+    ubyte[32] __glibc_reserved;
```

Comment:
This looks like the only platform specific part of this module here. Make a porters life easy and guard it properly with appropriate version blocks and static asserts.

i.e:
```
version (CRuntime_Glibc)
{
    version (X86_64)
    {
        struct aiocb
        {
            //... 
        }
    }
    else
        static assert(0);
}
else
    static assert(false, "Unsupported platform");
```